### PR TITLE
Add occ command to manage applicables

### PR DESCRIPTION
**Pull request by icewind1991**
Usage example: 

Also accepts to usual `--output` arguments to make it scriptable

### CODE REVIEW

File: apps/files_external/command/applicable.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		$type = $input->getArgument('type');
+		try {
+			$mount = $this->globalService->getStorage($mountId);
+		} catch (NotFoundException $e) {
+			$output->writeln('<error>Mount with id "' . $mountId . ' not found, check "occ files_external:list" to get available mounts"</error>');
+			return 404;
+		}
+
+		if ($type !== 'users' && $type !== 'groups') {
+			$output->writeln('<error>Invalid applicable type "' . $type . '"</error>');
```
Comment (MorrisJobke) [inline apps/files_external/command/applicable.php]: Give a hint that `users` and `groups` are valid options? ;)

**Comment by MorrisJobke** [comment]
What about unit tests?

File: apps/files_external/command/applicable.php (hunk 1 above)
Comment (icewind1991) [inline apps/files_external/command/applicable.php]: done

**Comment by icewind1991** [comment]
> What about unit tests?

Done

**Comment by PVince81** [comment]
Okay, I was having a weird UX because I used a different order:
```
sudo -u wwwrun ./occ files_external:applicable --add group1 1 groups
```

How about `--add-group group1 --add-user user` ? Would be more intuitive than having a dangling type argument ?

**Comment by PVince81** [comment]
I also suggest to add another option `--remove-all` to make it easier to switch back to "all", if that makes sense.

**Comment by PVince81** [comment]
Apart from that the code looks good.

**Comment by icewind1991** [comment]
Switched to separate `--add-user` and `-add-group` options

**Comment by PVince81** [comment]
@davitol another one :smile:

**Comment by PVince81** [comment]
- [ ] BUG: am able to add non-existing user/groups

**Comment by PVince81** [comment]
Otherwise this looks good!

**Comment by PVince81** [comment]
@icewind1991 can you fix the bug ?

**Comment by icewind1991** [comment]
> can you fix the bug ?

Done

**Comment by PVince81** [comment]
- [ ] BUG: I must not be able to add applicables to personal storages !

Best would be to make the API/backend deny this too!

**Comment by icewind1991** [comment]
> I must not be able to add applicables to personal storages !

Fixed

**Comment by PVince81** [comment]
:+1:

Needs a second reviewer @jmaciasportela @Xenopathic @SergioBertolinSG

**Comment by jmaciasportela** [comment]
:+1:  Good work !

**Pull request merged**